### DISCUSSION TITLE
Migration example simplification?

### DISCUSSION THREAD

**Discussion by mcarans** [sqlalchemy/sqlalchemy#9405] [Usage Questions] [answered]
This is in the [migration example](https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#migration-20-query-usage):
```
session.scalars(
  select(User).
  filter_by(name="some user").
  limit(1)
).first()
```

I was wondering if this would so the same thing?
```
session.scalar(
  select(User).
  filter_by(name="some user").
  limit(1)
)
```

Also can `limit(1)` be changed to first()?

**Comment by zzzeek** [comment, marked as answer]
in that specific case the two statements are the same.

removing limit(1) means the SELECT statement will deliver all rows to the client, which will then be thrown away.  typically wasteful but it the statement only returns one row anyway, it doesnt matter.

**Reply by mcarans** [reply]
Thanks!

```
session.scalar(
  select(User).
  filter_by(name="some user").
  first()
)
```
Would `first()` work as above instead of `limit(1)` or are they not equivalent?

**Reply by zzzeek** [reply]
there's no first() method on select()
